### ISSUE TITLE
Upgrading to Meteor 1.6.1 or newer when it works well enough. Try to remove fibers.

### ISSUE THREAD

**Issue by xet7** [wekan/wekan#1522]
Current problems with upgrading to 1.6.1:
- 100% CPU usage after upgrading to 1.6.1
- Upgrading caused [downgrading to broken version 1.0.x of coffeescript](https://github.com/meteor/meteor/issues/9609) . I don't know what causes that, because [peerlibrary:computed-field](https://github.com/peerlibrary/meteor-computed-field) v0.9.0 now uses Javascript, so no coffeescript dependency there anymore
- Downgrades could be because some packages still depend on older version of CoffeeScript. I have not yet checked do all packages support CoffeeScript 2.0, that would be the best option for upgrades.
- I did try some upgrades with i18n libraries that Wekan has currently, but did not get those working correctly yet.
- meteor-spk would need [new release for meteor 1.6.1](https://github.com/sandstorm-io/meteor-spk/issues/30)

**Comment by RobinELM** [comment]
I thought the 100%CPU issue would be solved by moving to Node 8.9 but apparently not the case ='(

On a VM with 6 vCPUs and around ~60 registered users, the application crash when more that 6 are working simultaneously ...
--> To which version would you advise to downgrade to avoid this issue ?

I though that the CPU usage issue has been spotted and solved but if it can be identified clearly I would be happy to support a bounty .. this stability problem is really killing any future usage :(

**Comment by xet7** [comment]
@RobinELM 

This is the recommended way to install Wekan, it's very fast and easy:
https://github.com/wekan/wekan-snap/wiki/Install

Current supported versions of Node, Mongo etc are listed at:
https://github.com/wekan/wekan/blob/devel/Dockerfile

Production setup for thousands of users with Docker at AWS:
https://github.com/wekan/wekan/wiki/AWS

Details about adding bounty to this issue: https://wekan.team
In short, copying this issue URL to bountysource.com search field fill find bounty URL.

Supporting this issue with bounties is very welcome, because I have already
spent very many hours trying to upgrade Meteor to 1.6.1, without
anyone supporting this financially.

**Comment by carlos22** [comment]
Will the upgrade to meteor 1.6.1 automatically bring support for mongo-db 3.4/3.6?

**Comment by xet7** [comment]
@carlos22 

I don't know, that needs to be tested, does it work with all dependencies of Wekan.

**Comment by xet7** [comment]
@carlos22 

Only Meteor 1.6.1 has support for newer version of MongoDB. Previous versions did not support 3.4 at all.

That MongoDB version issue is mainly about having attachments in MongoDB. There is #142 about storing attachments elsewhere, after that there should not be problems with newer MongoDB versions.

Support for other databases comes sometime in future in [Wekan Rust version](https://github.com/wekan/wekan.rs), but it's still in early architecture planning stages.

**Comment by carlos22** [comment]
Ok thanks for the updates :-) Keep the good work up.

**Comment by xet7** [comment]
@carlos22 

I don't yet have [enough funds for rewrite](https://github.com/wekan/wekan/wiki/FAQ#can-wekan-be-rewritten-in-another-programming-language) so I will continue Meteor version for now, and add other database support when it comes possible.

**Comment by xet7** [comment]
It would be nice to get rid of fibers sometime. Fibers is hard to compile on arm64 and vagrant, and also has 100% CPU usage bugs.

**Comment by ocdtrekkie** [comment]
I don't even know what fibers does. I just know that I hate it.

**Comment by xet7** [comment]
@ocdtrekkie 

[Fibers](https://www.npmjs.com/package/fibers) does coroutines / subroutines. So you can do some small stuff, pause it, go to other stuff, and pause/continue as needed. This is helpful, because Javascript usually is single threaded, so it makes multitasking possible. Fibers is made with C++, so it's hard to compile. In meteor.js you don't need to write word "Fibers", because Meteor uses it in background non-visible automatic way. Fibers is a hack, and a non-standard way to use node.js.

Better alternative is to change use [promises and async-await](https://blog.meteor.com/using-promises-and-async-await-in-meteor-8f6f4a04f998), it is standard way in node. But I don't know can Meteor and all included Wekan packages work completely without Fibers. @kentonv do you know?

**Comment by xet7** [comment]
TODO: try upgrading to 1.7
https://blog.meteor.com/meteor-1-7-and-the-evergreen-dream-a8c1270b0901

**Comment by xet7** [comment]
Edge now has Meteor 1.8.1-beta.0 and MongoDB 4.0.3 available for Docker testing https://github.com/wekan/wekan/blob/edge/CHANGELOG.md

**Comment by xet7** [comment]
Wekan uses Meteor 1.6.x. You can [test Meteor 1.8.x based Docker tags](https://github.com/wekan/wekan/issues/1947#issuecomment-435886419) and report are there speed improvements to that issue.
